pkg/ingester/ingester_claim.go

```diff
@@ -102,13 +106,25 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 		return err
 	}
 
+	i.joiningSampleQueueLock.Lock()
 	i.userStatesMtx.Lock()
-	defer i.userStatesMtx.Unlock()
+	i.userStates = userStates
+	i.userStatesMtx.Unlock()
+
+	for j := range i.joiningSampleQueue {
+		userSamples := &i.joiningSampleQueue[j]
+		userCtx := user.InjectOrgID(stream.Context(), userSamples.userID)
+		for k := range userSamples.samples {
+			i.append(userCtx, &userSamples.samples[k])
+		}
+	}
+	i.joiningSampleQueue = []userSamples{}
```

Comment:
Probably want to put this whole loop into a separate function and utilise a defer to do the unlock?